internal/subscriber/subscriber.go

```diff
@@ -410,36 +368,33 @@ func (s *Subscriber) Run(ctx context.Context) {
 		select {
 		case req := <-s.requestChan:
 			// on channel closure, we can get nil data
-			if req == nil {
-				continue
-			}
-			if ctx.Err() != nil {
+			if req == nil || ctx.Err() != nil {
```

Comment:
why is ctx.Err check needed, that is handlled by ctx.Done ?

Reply:
Was seeing a case where the loop enters the case block and then the ctx() gets canceled.